### PR TITLE
GLT-788: allow adding non-barcoded boxables to boxes

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxableRestController.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+@RequestMapping("/rest/boxables")
+
+public class BoxableRestController extends RestController {
+  @Autowired
+  private BoxService boxService;
+
+  @RequestMapping(value = "/search", method = RequestMethod.GET, produces = "application/json")
+  @ResponseBody
+  public List<BoxableDto> search(@RequestParam("q") String search, HttpServletRequest request, HttpServletResponse response,
+      UriComponentsBuilder uriBuilder) throws IOException {
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxableRestController.java]: MINOR Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by sqwci** [comment]
SonarQube analysis reported 4 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 2 critical
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [box_visualization.js#L31](https://github.com/TGAC/miso-lims/blob/1d6c6f1cb489ac2f26e9fc77dc0fcc630a831146/miso-web/src/main/webapp/scripts/box_visualization.js#L31): Add the "let", "const" or "var" keyword to this declaration of "isMultiSelectClick" to make it explicit. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2703)
1. ![CRITICAL][CRITICAL] [Dtos.java#L423](https://github.com/TGAC/miso-lims/blob/1d6c6f1cb489ac2f26e9fc77dc0fcc630a831146/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L423): Refactor this method to reduce its Cognitive Complexity from 24 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [HibernateBoxDao.java#L43](https://github.com/TGAC/miso-lims/blob/1d6c6f1cb489ac2f26e9fc77dc0fcc630a831146/sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateBoxDao.java#L43): Define a constant instead of duplicating this literal "alias" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [box_visualization.js#L31](https://github.com/TGAC/miso-lims/blob/1cac8ff300ba7cb12bf82c8ef3cde095f85bbcdf/miso-web/src/main/webapp/scripts/box_visualization.js#L31): Add the "let", "const" or "var" keyword to this declaration of "isMultiSelectClick" to make it explicit. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2703)
1. ![CRITICAL][CRITICAL] [Dtos.java#L423](https://github.com/TGAC/miso-lims/blob/1cac8ff300ba7cb12bf82c8ef3cde095f85bbcdf/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L423): Refactor this method to reduce its Cognitive Complexity from 24 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [HibernateBoxDao.java#L43](https://github.com/TGAC/miso-lims/blob/1cac8ff300ba7cb12bf82c8ef3cde095f85bbcdf/sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateBoxDao.java#L43): Define a constant instead of duplicating this literal "alias" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

File: miso-web/src/main/webapp/scripts/box_visualization.js [hunk 2]
```diff
@@ -93,17 +93,24 @@ var BoxPosition = function(opts) {
...
+      return;
+    }
     self.selectedItems.forEach(function(item) {
       item.unselect()
     });
     self.selectedItems = [];
   };
 
   self.select = function(item, skipCallback) {
+    if (disabled) {
```
Comment (apmasell) [inline miso-web/src/main/webapp/scripts/box_visualization.js]: :+1:

**Pull request merged**